### DISCUSSION TITLE
Branch Protection Rule "Require approval of the most recent push" should provide option to ignore clean/empty commits

### DISCUSSION THREAD

**Discussion by zepeng811** [community/community#41547] [Pull Requests] [answered]
### Select Topic Area

Product Feedback

### Body

Recently GitHub implemented a branch protection rule feature [Require approval from someone other than the last pusher](https://github.blog/changelog/2022-10-20-new-branch-protections-last-pusher-and-locked-branch/#:~:text=Require%20approval%20from%20someone%20other%20than%20the%20last%20pusher&text=Meaning%2C%20the%20most%20recent%20user,will%20also%20need%20to%20approve.). This is a great feature to prevent developers adding code to someone else's branch/PR and approving their own changes, however we are experiencing an issue when trying to enable this feature in our organization.

Our organization requires the branch to be up to date with the protected branch before merging (The nested feature under required status check):
![image](https://user-images.githubusercontent.com/107065343/207129013-f8aff198-c211-47e5-ab7a-5c656e2b45c1.png)

Which enabling both rules causes a big problem for our workflow:
1. PR Author opens the PR
2. reviewer approves the change
3. the branch is out of date with the protected branch, either the reviewer or the PR author clicks the "Update Branch" button to sync the branch with the protected branch. Which it pushes an empty/clean commit with no code changes.
   * if the PR author pushed that empty/clean commit, the previous approval is invalidated. The reviewer must approve the PR again
   * if the reviewer pushed that empty/clean commit, the previous approval is also invalidated and the current reviewer cannot re-approve the change. It must find another reviewer for the approval.

I think it makes sense for this particular rule to ignore (or at least provide the option to ignore) clean/empty commits pushed to the branch. Let me know if the above workflow does not make sense or additional details are required.

**Comment by dgteixeira** [comment]
Hey, 
We are suffering from the same problem.

Currently, we had to turn off the `last_pusher` because we were having way too many problems with that workflow.

You still had no feedback from the GitHub team?

**Reply by zepeng811** [reply]
Hey,

yeah unfortunately I haven't heard any feedback from anyone on the GitHub team, we also had to turn off the feature since I reported the problem.

**Reply by dgteixeira** [reply]
Ok, thanks for the feedback! I opened a support case with them and linked this discussion!
Let's see if it gets us some feedback.

**Comment by patrick-knight** [comment, marked as answer]
Hey y'all! 

We made a number of [substantial changes to pull requests earlier this month ](https://github.blog/changelog/2023-06-06-security-enhancements-to-required-approvals-on-pull-requests/)which included updated to "last pusher" and this behavior.
